app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -260,3 +261,26 @@ export function* unpinTab({ pageId, webContentsId }) {
   yield put(PageActions.setPageUnpinned(pageId));
   yield apply(ipcRenderer, ipcRenderer.send, ['guest-pinned-state', webContentsId, false]);
 }
+
+export function* restoreClosedPage() {
+  const closedPages = yield select(PageSelectors.getRecentlyClosedPages);
+  const lastPage = closedPages.last();
+  if (!lastPage) {
+    return;
+  }
+
+  const ancestor = lastPage.id;
+  const location = lastPage.restoreURL;
+  const pageIndex = lastPage.pageIndex;
+
+  const action = PageEffects.createPageSession({
+    location,
+    // TODO actually wire createPageSession to use ancestor
```

Comment:
Was wiring this up last night, looks like an issue in UAS will need to be filed too (will add in comment)